components/payments/settings/payment-webhooks.tsx

```diff
@@ -7,5 +21,210 @@ interface PaymentsWebhooksProps {
 export const PaymentsWebhooks: React.FC<PaymentsWebhooksProps> = ({
   accountId,
 }) => {
-  return <Flex flexDir="column" gap={8}></Flex>;
+  const { data: webhookApiKey, isLoading: isLoadingSecretKey } =
+    usePaymentsWebhooksSecretKeyByAccountId(accountId);
+  const {
+    data: webhooks,
+    isLoading: isGetLoading,
+    isFetched,
+  } = usePaymentsWebhooksByAccountId(accountId);
+  const { mutate: updateWebhook, isLoading: isUpdateLoading } =
+    usePaymentsUpdateWebhook(accountId);
+  const { mutate: createWebhook, isLoading: isCreateLoading } =
+    usePaymentsCreateWebhook(accountId);
+
+  const toast = useToast();
+
+  const triggerAlert = (
+    status: "error" | "success",
+    header: string,
+    description: string,
+  ) => {
+    toast({
+      position: "bottom",
+      variant: "solid",
+      title: header,
+      description,
+      status,
+      duration: 9000,
+      isClosable: true,
+    });
+  };
+
+  const { productionWebhooks, testnetWebhooks } = useMemo(() => {
+    if (webhooks) {
+      const productionWebhooks = webhooks.filter(
+        (webhook) => webhook.isProduction,
+      );
+      const testnetWebhooks = webhooks.filter(
+        (webhook) => !webhook.isProduction,
+      );
+
+      return { productionWebhooks, testnetWebhooks };
+    }
+    return { productionWebhooks: [], testnetWebhooks: [] };
+  }, [webhooks]);
+
+  const onUpdateWebhook: PaymentsWebhooksTableProps["onUpdate"] = (
+    webhook,
+    newUrl,
+  ) => {
+    if (!isValidWebhookUrl(newUrl)) {
```

Comment:
can't we do this validation in the form itself? I'm doing something similar for successCallbackUrl inn create-update--checkout-button.tsx. You could copy the component as it or abtract it, (line 702-731)

It's always better if we can give feedback to the user _before_ they click a button to do something